systemd/emulation.go

```diff
@@ -98,12 +99,22 @@ func (s *emulation) AddMountUnitFile(snapName, revision, what, where, fstype str
 		return "", fmt.Errorf("bind-mounted directory is not supported in emulation mode")
 	}
 
-	mountUnitName, actualFsType, options, err := writeMountUnitFile(snapName, revision, what, where, fstype)
```

Comment:
if I understand emulation doesn't understand `what` being a directory anymore, we should error on that, I suppose is not something we need to support anyway

Reply:
We do error on that already right at the beginning of AddMountUnitFile; are you suggesting to check it earlier?

Reply:
ah, sorry, I missed that because it was preexisting and wasn't clearly in the diff. No it's fine.